### PR TITLE
Services actuator, revisited

### PR DESCRIPTION
**Pull request by bart-vmware**
## Description

Adds a services actuator to view the contents of the .NET IoC container. This mirrors the behavior of the beans actuator for tools like SpringBootAdmin. Continuation of the work from #1129.

Related to #668.

## Quality checklist

- [x] Your code complies with our [Coding Style](https://github.com/SteeltoeOSS/.github/blob/main/contributing-docs/contributing-code-style.md).
- [x] You've updated unit and/or integration tests for your change, where applicable.
- [x] You've updated documentation for your change, where applicable.
      If your change affects other repositories, such as [Documentation](https://github.com/SteeltoeOSS/Documentation), [Samples](https://github.com/SteeltoeOSS/Samples) and/or [MainSite](https://github.com/SteeltoeOSS/MainSite), add linked PRs here.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.
- [x] You've added required license files and/or file headers (explaining where the code came from with proper attribution), where code is copied from StackOverflow, a blog, or OSS.

### CODE REVIEW

File: src/Management/src/Endpoint/Services/ServiceRegistration.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    public ISet<string> Dependencies { get; }
+
+    public ServiceRegistration(ServiceDescriptor descriptor)
+    {
+        ArgumentGuard.NotNull(descriptor);
+
+        Name = descriptor.ServiceType.Name;
+        Scope = descriptor.Lifetime.ToString();
+        Type = descriptor.ImplementationInstance?.GetType().ToString() ?? descriptor.ServiceType.ToString();
+        AssemblyName = descriptor.ServiceType.AssemblyQualifiedName!;
```
Comment (bart-vmware) [inline src/Management/src/Endpoint/Services/ServiceRegistration.cs:37]: These values may be incorrect, compared to how Spring works. See related conversation at https://github.com/SteeltoeOSS/Steeltoe/pull/1129#discussion_r1377424010.

File: src/Management/src/Endpoint/ManagementHostBuilderExtensions.cs [hunk 2]
```diff
@@ -329,6 +330,23 @@ public static IHostBuilder AddTraceActuator(this IHostBuilder hostBuilder)
         return AddTraceActuator(hostBuilder, MediaTypeVersion.V2);
     }
 
+    /// <summary>
+    /// Adds the Services actuator to the application.
+    /// </summary>
+    /// <param name="hostBuilder">
+    /// Your HostBuilder.
+    /// </param>
```
Comment (bart-vmware) [inline src/Management/src/Endpoint/ManagementHostBuilderExtensions.cs:338]: Documentation should be improved, as suggested at https://github.com/SteeltoeOSS/Steeltoe/pull/1129#discussion_r1376307264. Postponing until #1223 has been merged, as it shuffles things around.

**Review by TimHess — APPROVED** [review]
LGTM

**Pull request merged**